packages/tx/src/baseTransaction.ts

```diff
@@ -169,21 +170,43 @@ export abstract class BaseTransaction<TransactionObject> {
     return stringError ? errors : errors.length === 0
   }
 
+  protected _validateYParity() {
+    const { v } = this
+    if (v !== undefined && v !== BigInt(0) && v !== BigInt(1)) {
+      const msg = this._errorMsg('The y-parity of the transaction should either be 0 or 1')
+      throw new Error(msg)
+    }
+  }
+
+  /**
+   * EIP-2: All transaction signatures whose s-value is greater than secp256k1n/2are considered invalid.
+   * Reasoning: https://ethereum.stackexchange.com/a/55728
+   */
+  protected _validateHighS() {
+    const { s } = this
+    if (this.common.gteHardfork('homestead') && s !== undefined && s > SECP256K1_ORDER_DIV_2) {
+      const msg = this._errorMsg(
+        'Invalid Signature: s-values greater than secp256k1n/2 are considered invalid'
+      )
+      throw new Error(msg)
+    }
+  }
```

Comment:
To clarify: are these checks which initiallly have been missing or is this some kind of refactor?

Reply:
Update: Ah, just seeing that this is taken out of the inherited classes.

So `_validateHighS()` is ok to move out I guess but `_validateYParity()` should not be taken out since this doesn't go for the legacy tx and in base tx we only want to have code which is really common for all three existing tx types.

Side note: this whole inheritance structure for txs with tx types is not optimal. We have tried to come up with some - more modular - replacement but weren't satisfied yet on the results. If you want you can give this some initial thought. That would be a somewhat big refactor though, not sure if the scope on this would be too large to get into the coming breaking release round.

Reply:
What's your suggesting regarding yParity validation? It's not great to have it duplicated in 3 files. The error names are the same, as is the logic. Having one different behavior somewhere is error-prone, we can forget to update all 3 files. 

As for class inheritance, check out micro-eth-signer, which implements all txs in 200 locs:

https://github.com/paulmillr/micro-eth-signer/blob/727d822030cbb5ba7e542a3a4ec6d40504069f82/index.ts#L314

Reply:
Let's please not try to solve this on the sideline here, in doubt open a small issue on this.

Reply:
> As for class inheritance, check out micro-eth-signer, which implements all txs in 200 locs:
> 
> https://github.com/paulmillr/micro-eth-signer/blob/727d822030cbb5ba7e542a3a4ec6d40504069f82/index.ts#L314

We can talk about the relationship of libraries like `micro-eth-signer` and `@ethereumjs/tx`, but this will likely take some time. 🙂